compiler/crates/relay-compiler/src/main.rs

```diff
@@ -0,0 +1,56 @@
+use env_logger::Env;
+use log::{error, info};
+use relay_compiler::{compiler::Compiler, config::Config};
+use std::{path::PathBuf, sync::Arc};
+use structopt::StructOpt;
+
+#[derive(StructOpt)]
+#[structopt(
+    name = "Relay Compiler",
+    about = "Compiler to produce Relay generated files."
+)]
+struct Opt {
+    /// Compile and watch for changes
+    #[structopt(long, short)]
+    watch: bool,
+
+    /// Path to the compiler config file
+    config: PathBuf,
+
+    /// Root directory of the project
+    root: PathBuf,
+}
+
+#[tokio::main]
+async fn main() {
+    env_logger::from_env(Env::default().default_filter_or("info, error, warn")).init();
+
+    let opt = Opt::from_args();
```

Comment:
Do we print some sort of "debug" info here, like [maybe the version that is running](https://github.com/clap-rs/clap/blob/master/README.md#L215-L217) etc...

Could potentially use [Clap](https://github.com/clap-rs/clap)

Granted this is just a "initial start"—but think we can definitely jazz it up!

Reply:
This is using https://crates.io/crates/structopt which is a wrapper around Clap.

This should already generate a `--help` command and I think a `--version` based on the Cargo.toml.